setyaev_alexander/graph.cpp

```diff
@@ -0,0 +1,240 @@
+  EdgeId get_new_edge_id() const;
+
+  Vertex& get_vertex(const VertexId& id);
+
+  VertexId vertex_id_counter_ = 0;
+  EdgeId edge_id_counter_ = 0;
+};
+
+VertexId Vertex::get_id() const {
+  return id_;
+}
+EdgeId Edge::get_id() const {
+  return id_;
+}
+
+VertexId Edge::get_first_id() const {
+  return first_vertex_id_;
+}
+
+VertexId Edge::get_second_id() const {
+  return second_vertex_id_;
+}
+
+bool Edge::operator==(const Edge& right_edge) const {
+  if (this->get_id() == right_edge.get_id()) {
+    return true;
+  }
+  return false;
+}
+
+bool Vertex::operator==(const Vertex& right_edge) const {
+  if (this->get_id() == right_edge.get_id()) {
+    return true;
+  }
+  return false;
+}
+
+void Graph::add_vertex() {
+  vertices_.emplace_back(get_new_vertex_id());
+  vertex_id_counter_++;
+}
+
+VertexId Graph::get_new_vertex_id() const {
+  return vertex_id_counter_;
+}
+EdgeId Graph::get_new_edge_id() const {
+  return edge_id_counter_;
+}
+
+Vertex& Graph::get_vertex(const VertexId& id) {
+  for (auto& vertex : vertices_) {
+    if (vertex.get_id() == id) {
+      return vertex;
+    }
+  }
+  throw std::runtime_error("Vertex not found!\n");
+}
+
+EdgeId Vertex::get_connected_edge_id(int idx) const {
+  return connected_edge_ids_[idx];
+}
+
+void Vertex::add_connected_edge_id(EdgeId id) {
+  connected_edge_ids_.push_back(id);
+}
+int Vertex::get_connected_edges_count() const {
+  return connected_edge_ids_.size();
+}
+
+void Graph::add_edge(VertexId first_vertex_id, VertexId second_vertex_id) {
+  const auto& new_edge =
+      edges_.emplace_back(get_new_edge_id(), first_vertex_id, second_vertex_id);
+  auto& first_vertex = get_vertex(first_vertex_id);
+  first_vertex.add_connected_edge_id(new_edge.get_id());
+  auto& second_vertex = get_vertex(second_vertex_id);
+  second_vertex.add_connected_edge_id(new_edge.get_id());
+
+  edge_id_counter_++;
+}
+
+Graph generate_graph() {
+  Graph graph;
+
+  for (int i = 0; i < VERTICES_COUNT(); ++i) {
+    graph.add_vertex();
+  }
+
+  graph.add_edge(0, 1);
+  graph.add_edge(0, 2);
+  graph.add_edge(0, 3);
+  graph.add_edge(1, 4);
+  graph.add_edge(1, 5);
+  graph.add_edge(1, 6);
+  graph.add_edge(2, 7);
+  graph.add_edge(2, 8);
+  graph.add_edge(3, 9);
+  graph.add_edge(4, 10);
+  graph.add_edge(5, 10);
+  graph.add_edge(6, 10);
+  graph.add_edge(7, 11);
+  graph.add_edge(8, 11);
+  graph.add_edge(9, 12);
+  graph.add_edge(10, 13);
+  graph.add_edge(11, 13);
+  graph.add_edge(12, 13);
+
+  return graph;
+}
+
+std::string Graph::to_string() const {
+  std::stringstream json;
+
+  json << "{" << std::endl << "	\"vertices\": [" << std::endl;
+
+  for (const auto& vertex : vertices_) {
+    json << "		{\n			\"id\": " << vertex.get_id()
+         << "," << std::endl
+         << "			\"edge_ids\": [";
+    for (int j = 0; j < vertex.get_connected_edges_count(); ++j) {
+      json << vertex.get_connected_edge_id(j);
+      if (j < vertex.get_connected_edges_count() - 1) {
```

Comment:
Чем плодить миллион функций, лучше просто верни `const` ссылку на вектор
```suggestion
    const auto& edge_ids = vertex.connected_edge_ids();
    for (const auto edge_id : edge_ids)  {
      json << edge_id;
      if (edge_id != edge_ids.back()) {
```